spec/core/ics-004-channel-and-packet-semantics/README.md

```diff
@@ -332,7 +332,7 @@ function chanOpenTry(
          previous.counterpartyPortIdentifier === counterpartyPortIdentifier &&
          previous.counterpartyChannelIdentifier === "" &&
          previous.connectionHops === connectionHops &&
-         previous.version === version)
+         previous.version === counterpartyVersion
```

Comment:
For simplicity, we enforce that in case of crossing hellos, previous version must be exactly same as counterparty version.

We could relax this requirement by passing the previous version into the callback, and letting the callback decide if they are compatible. However this is a more complicated solution to support a rather small edgecase, so it will not be supported for now.

Reply:
I have a question about this whole "crossing hellos" logic.

I believe that made sense when the relayer selected the channel, and it was possible to init the exact same channel on both sides, which then try against each other.

With the channel auto-gen, there is no possibility that an init and a try on the same chain create the same (port, channel) identifier.

I think this case can be removed, which will simplify the code.

Reply:
Please explain to me where in practice this case can still occur. I may be missing something

Reply:
Bump. 

I think this concept is no longer possible with auto-assigned channel_ids.
Can someone please verify this or correct me if I am wrong (and explain how this case occurs).

If we are cleaning up specs, removing or explaining this case would be a huge help.

Reply:
I think the existence of "crossing hellos" scenarios now acts as an optimization as opposed to a safety mechanism.

We have chainA and chainB. For some reason both chains want to open a channel with X set of parameters. 

chainA calls ChanOpenInit with X parameters (counterparty channel identifier unspecified)
chainB call ChanOpenInit with X parameters (counterparty channel identifier unspecified)

In the existing specification, a relayer could notice both sides wish to create the same channel. It can call ChanOpenTry specifying the existing counterparty channel identifier, thus connecting the two sides without creating additional channel identifiers.

If the case above is removed, a new channel identifier/end would need to be created on both chains to complete the handshakes even if both chains created channels which are compatible. 

To the best of my knowledge, there aren't any attacks possible if we removed crossing hello support (I really haven't thought about it much).

Given that probability of two chains wanting to open the same channel in the same moment is very low, I think it makes sense to explicitly disallow "crossing hellos". A fair amount of our bugs have come from crossing hello edge cases. IBC applications are currently required to understand the "crossing hello" concept (in relation to claiming channel capability). Crossing hellos makes onboarding onto handshake logic more difficult. 

cc @cwgoes

Reply:
Okay, this is an optimisation for an edge case that makes the main case more complex to implement and bug-prone?

I agree with explicitly removing support for this. I mean, if no one uses it and no one outside of IG and Chris really understand it...

Reply:
Agreed, simply removing this code block on TRY and generating new channel identifier each time